File: src/main/java/com/ironhack/pokemon/model/Trainer.java
package com.ironhack.pokemon.model;

import javax.persistence.*;

@Entity
public class Trainer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String hobby;
    private String photo;

//    Constructor
    public Trainer() {
    }

    public Trainer(String name, String hobby, String photo) {
        this.name = name;
        this.hobby = hobby;
        this.photo = photo;
    }

//    Getter and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getHobby() {
        return hobby;
    }

    public void setHobby(String hobby) {
        this.hobby = hobby;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }
}

File: pokemon/src/app/trainer/classes/trainer.ts
export class Trainer {

    constructor(
        private _id: number,
        private _name: string,
        private _age: number,
        private _hobby: string,
        private _photo: string
    ){}

    
    public get id() : number {
        return this._id;
    }
    
    public get name() : string {
        return this._name;
    }
    
    public get age() : number {
        return this._age;
    }
   
    public get hobby() : string {
        return this._hobby;
    }
    
    public get photo() : string {
        return this._photo;
    }

    
    public set id(v : number) {
        this._id = v;
    }
    
    public set name(v : string) {
        this._name = v;
    }
    
    public set age(v : number) {
        this._age = v;
    }
    
    public set hobby(v : string) {
        this._hobby = v;
    }
    
    public set photo(v : string) {
        this._photo = v;
    }


}
